pandas/_libs/src/vendored/ujson/python/objToJSON.c

```diff
@@ -1610,6 +1610,11 @@ void Object_beginTypeContext(JSOBJ _obj, JSONTypeContext *tc) {
                                   PyArray_DescrFromType(NPY_DOUBLE));
         tc->type = JT_DOUBLE;
         return;
+    } else if (PyArray_IsScalar(obj, LongDouble)) {
```

Comment:
This works, but I think the lines directly following it are supposed to be a catchall for unsupported types. Do you know why that isn't being hit? I would rather we keep this generic instead of having to specify an error message for every type that we don't serialize

Reply:
There is a code section at the end of this function: 
```    tc->type = JT_OBJECT;
    pc->iterBegin = Dir_iterBegin;
    pc->iterEnd = Dir_iterEnd;
    pc->iterNext = Dir_iterNext;
    pc->iterGetValue = Dir_iterGetValue;
    pc->iterGetName = Dir_iterGetName;
    return;
```
By default everything falls to this section and this causes infinite loop.

Reply:
Right but I'm asking about the next branch after what you've added. 

```c
} else if (PyArray_Check(obj) && PyArray_CheckScalar(obj)) {
```

Do you know what hits that currently? From reading the function I think the intent of that was to generically catch the issue you've described, but its possible the invariant is incorrect. Would something pass both `PyArray_Check` and `PyArray_CheckScalar`? Maybe the `PyArray_Check` call is incorrect and removing that alone would fix your issue?

Reply:
Any value like: np.array(1) evaluates true for both of them. However yes, simply removing pyArray_check solves the issue. Again Thanks for the suggestion. Will updating the PR shortlyl.